src/NotepadNext/docks/SearchResultsDock.cpp

```diff
@@ -210,3 +211,18 @@ void SearchResultsDock::updateSearchStatus()
     if (currentFile)
         currentFile->setText(0, QStringLiteral("%1 (%L2 hits)").arg(currentFilePath).arg(totalFileHitCount));
 }
+
+void SearchResultsDock::on_pb_copyResults_released()
+{
+    QClipboard *clipboard = QGuiApplication::clipboard();
+    QString string;
+
+    QTreeWidgetItemIterator it(ui->treeWidget);
+    while (*it) {
+        string.append((*it)->text(0) + ' ' + (*it)->text(1) + '\n');
```

Comment:
Is there a more efficient way to build this string? In theory the treeWidget might contain thousands of items. Maybe it is a non-concern, just curious is all. Maybe even something like QString().arg() etc would be more efficient?

I'm not sure.  Haven't looked into this myself yet. Just curious is all.

Reply:
I haven't found any way to copy the tree contents without walking it, but as always, open to suggestions from more knowledgable people :)

Reply:
I'm not sure either. Was just curious. I'll see if there is anything else obvious.